modules/sr/robot/encoder.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from typing import List
+
+from controller import Robot, PositionSensor
+from shared_utils import RobotType
+from sr.robot.utils import get_robot_device
+
+
+def init_encoder_array(webot: Robot, robot_type: RobotType) -> List[Encoder]:
+    if robot_type == RobotType.FORKLIFT:
+        return [
+            RotaryEncoder(webot, 'left wheel sensor'),
+            RotaryEncoder(webot, 'right wheel sensor'),
+
+            LinearEncoder(webot, 'left gripper sensor'),
+            LinearEncoder(webot, 'right gripper sensor'),
+        ]
+    else:
+        return [
+            LinearEncoder(webot, 'bridge position sensor'),
+            LinearEncoder(webot, 'trolley position sensor'),
+            LinearEncoder(webot, 'hoist position sensor'),
+        ]
+
+
+class Encoder:
+    def __init__(self, webot: Robot, sensor_name: str):
+        self._encoder = get_robot_device(webot, sensor_name, PositionSensor)
+        self._encoder.enable(int(webot.getBasicTimeStep()))
+
+    @property
+    def rotation(self) -> float:
+        raise NotImplementedError("This sensor does not support measuring rotations.")
+
+    @property
+    def displacement(self) -> float:
+        raise NotImplementedError(
+            "This sensor does not support measuring linear displacement.",
```

Comment:
Any reason to use a base class here over just implementing natively? Omitting the methods entirely would likely yield more useful errors, as "This" in the context of this message could be ambiguous

Reply:
I had misremembered and thought the default error message was unhelpful, I've removed the custom errors in 1cce8a68ee85b4fd5c41e256b375ffd8c1dbe7bf